test/tests/smoke_test.rs

```diff
@@ -14,13 +14,25 @@ fn assert_output_contains(haystack: &str, needle: &str) {
 
 #[test]
 fn smoke_test() {
-    let rom = caliptra_builder::build_firmware_rom(&ROM_WITH_UART).unwrap();
-    let image = caliptra_builder::build_and_sign_image(
-        &FMC_WITH_UART,
-        &APP_WITH_UART,
-        ImageOptions::default(),
-    )
-    .unwrap();
+    #[cfg(not(feature = "fpga_realtime"))]
+    let (rom, fmc, app) = { (ROM_WITH_UART, FMC_WITH_UART, APP_WITH_UART) };
+
+    #[cfg(feature = "fpga_realtime")]
+    let (rom, fmc, app) = {
+        let mut rom_copy = ROM_WITH_UART;
+        let mut fmc_copy = FMC_WITH_UART;
+        let mut app_copy = APP_WITH_UART;
+
+        rom_copy.features = &["emu fpga_realtime"];
```

Comment:
Shouldn't this be &["emu", "fpga_realtime"]?

Reply:
It should have been, I don't know why this wasn't a problem until pulling in your preprocessor patch.
This goes away with moving to the builder.